pkg/app/piped/cmd/piped/piped.go

```diff
@@ -515,3 +526,52 @@ func (p *piped) sendPipedMeta(ctx context.Context, client pipedservice.Client, c
 
 	return err
 }
+
+// insertLoginUserToPasswd adds the logged-in user to /etc/passwd.
+// It requires nss_wrapper (https://cwrap.org/nss_wrapper.html)
+// to get the operation done.
+//
+// This is a workaround to deal with OpenShift less than 4.2
+// See more: https://github.com/pipe-cd/pipe/issues/1905
+func (p *piped) insertLoginUserToPasswd(ctx context.Context) error {
+	var stdout, stderr bytes.Buffer
+
+	// Use the id command so that it gets proper ids even in pure Go.
+	cmd := exec.CommandContext(ctx, "id", "-u")
+	cmd.Stdout = &stdout
+	cmd.Stderr = &stderr
+	if err := cmd.Run(); err != nil {
+		return fmt.Errorf("failed to get uid: %s", &stderr)
+	}
+	uid := strings.TrimSpace(stdout.String())
+
+	stdout.Reset()
+	stderr.Reset()
+
+	cmd = exec.CommandContext(ctx, "id", "-g")
+	cmd.Stdout = &stdout
+	cmd.Stderr = &stderr
+	if err := cmd.Run(); err != nil {
+		return fmt.Errorf("failed to get gid: %s", &stderr)
+	}
+	gid := strings.TrimSpace(stdout.String())
+
+	home, err := os.UserHomeDir()
+	if err != nil {
+		return fmt.Errorf("failed to detect the current user's home directory: %w", err)
+	}
```

Comment:
All of these can be replaced by using the standard `Current` function.
https://golang.org/pkg/os/user/#Current

Reply:
Current requires cgo or `$USER` set in the environment variable if the user isn't in `/etc/passwd`. At this time, it isn't in passwd yet, that's why I settled on using `id` command to get it.

Reply:
Kind of like this:
```bash
❯ docker run -v /Users/you/src/hoge/foo:/go/foo -u 2020:0 --rm -it golang:1-alpine sh
/go $ ./foo
2021/05/13 06:02:42 user: Current requires cgo or $USER set in environment
```

Reply:
👍